### PR TITLE
Better error messages for extension constructor type mismatches

### PR DESCRIPTION
**Pull request by trefis**
A bootstrap was required because all the idents are shifted because the code was previously creating a fresh ident with name `""` at a frequent rate.

### CODE REVIEW

**Comment by gasche** [comment]
I think the current bootstrap policy is that the bootstrap changes should come on their own?

File: testsuite/tests/typing-modules/Test.ml [hunk 1]
```diff
@@ -137,10 +137,11 @@ Error: Signature mismatch:
          sig type t += E of int end
        is not included in
          sig type t += E end
-       Extension declarations do not match:
```
Comment (gasche) [inline testsuite/tests/typing-modules/Test.ml]: This seems to be a regression.

File: typing/includecore.ml [hunk 2]
```diff
@@ -364,21 +362,23 @@ let extension_constructors ~loc env ~mark id ext1 ext2 =
...
     Btype.newgenty (Tconstr(ext2.ext_type_path, ext2.ext_type_params, ref Mnil))
   in
-  if Ctype.equal env true
-       (ty1 :: ext1.ext_type_params)
-       (ty2 :: ext2.ext_type_params)
-  then
-    if compare_constructor_arguments ~loc env (Ident.create "")
+  if
+    not (Ctype.equal env true (ty1 :: ext1.ext_type_params)
+           (ty2 :: ext2.ext_type_params))
```
Comment (gasche) [inline typing/includecore.ml]: The previous alignment of `1` and `2` parameters was nicer to read.

**Review by gasche — COMMENTED** [review]
The change looks correct to me, but see minor comments.

File: testsuite/tests/typing-modules/Test.ml (hunk 1 above)
Comment (trefis) [reply testsuite/tests/typing-modules/Test.ml]: Indeed, typo on my part. I'll push a fix now.

**Comment by trefis** [comment]
> I think the current bootstrap policy is that the bootstrap changes should come on their own?

I thought the policy was that each commit should compile.
If I put the bootstrap on its own then the compiler doesn't compile.

**Comment by gasche** [comment]
```
Checking e42a2e4e8ed0c7b5ccf5f05083a9c53f803bc33b: testsuite/tests/typing-modules/Test.ml
./testsuite/tests/typing-modules/Test.ml:180.81: [long-line] line is over 80 columns
```

You need to break a line in

```ocaml
 module M : sig type t += E of { x : int } end = struct type t += E of int end;;
```

**Review by gasche — APPROVED** [review]
Good to merge when CI passes.

(I still think that the bootstrap commit should be separate, but I won't fight for it.)

**Comment by trefis** [comment]
Raah, I really need to setup up that precommit hook...
Anyway, I amended the commit to fix the issue and pushed again.

**Pull request merged**